02.wo_Defects/memory_allocation_failure.c

```diff
@@ -258,7 +258,7 @@ void memory_allocation_failure_007 ()
 */
 enum {max_buffer = MAX_VAL*2};
 char * memory_allocation_failure_008_func_001 (const char *msg) {
-  const char *error_log = msg;
+  char *error_log = (char *)msg;
```

Comment:
this is a funny one. snprintf %s reads through an lvalue of type "char *" which "const char *" is not effectively compatible with. So we need to drop the const qualifier before calling it.